### PR TITLE
move vue and vue-calendar-heatmap to npm/webpack

### PR DESCRIPTION
**Pull request by silverwind**
- unvendor vue and vue-calendar-heatmap
- remove unused moment.js leftover from previous heatmap version
- ensure webpack loads the full version of vue
- fix vue devmode warning related to 'searchLimit' type
- update vue from 2.6.6 to 2.6.11

I wanted to name the chunk heatmap.js but adblockers don't like that filename [1].

[1] https://github.com/easylist/easylist/blob/3899d5dff33216c0bc64f09ff15d376f346d3e33/easyprivacy/easyprivacy_general.txt#L2095

### CODE REVIEW

**Comment by silverwind** [comment]
There are more Vue warnings visible when webpack builds in development mode. I want to eventually enable us to use that mode, but right now I can not get bindata-less builds to work at all for some reason which is a requirement for that.

**Comment by silverwind** [comment]
The heatmap should probably be moved to a `.vue` file at some point and loaded via `vue-loader` but I don't have the motivation to do that as I'm not that much of a Vue fan 😉.

File: web_src/js/features/userHeatmap.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    delimiters: vueDelimeters,
+    el,
+
+    data: {
+      suburl: document.querySelector('meta[name=_suburl]').content,
+      heatmapUser: window.config.heatmapUser,
+      locale: {
+        contributions: 'contributions',
+        no_contributions: 'No contributions',
+      },
```
Comment (silverwind) [inline web_src/js/features/userHeatmap.js:95]: Note: These were previously also "hardcoded". There was a `locale` argument but it was unused so the defaults were always used. We probably need a way to expose localization values to JS later.

**Comment by silverwind** [comment]
@lafriks any feedback? If you like, I can try porting heatmap to a `.vue` file but would prefer if you do it because vue is black magic to me. I think we can probably streamline the vue initialization parts to a shared vue rendering function.

**Comment by lafriks** [comment]
Looks good, it can be converted to Vue in other pr

**Comment by silverwind** [comment]
Rebased. This is good to go from my perspective.

**Review by guillep2k — APPROVED** [review]

**Comment by gary-kim** [comment]
> The heatmap should probably be moved to a `.vue` file at some point and loaded via `vue-loader` but I don't have the motivation to do that as I'm not that much of a Vue fan wink.

Okay, I can do that part. Added to todo.

**Comment by silverwind** [comment]
@gary-kim I'd prefer if we create a single vue rendering function to be used by all the vue components. I think the current code initializes vue in 2 or 3 different places, which seems like useless code duplication to me. Maybe create `src/vue.js` and export that rendering function there.

**Comment by gary-kim** [comment]
I like that idea. Make a single rendering function that imports all the Vue components and calls `$mount` on them.

**Comment by silverwind** [comment]
Rebased

**Comment by silverwind** [comment]
Any more reviews here?

File: webpack.config.js [hunk 2]
```diff
@@ -200,5 +200,8 @@ module.exports = {
   },
   resolve: {
     symlinks: false,
+    alias: {
+      vue$: 'vue/dist/vue.esm.js', // needed because vue's default export is the runtime only
```
Comment (gary-kim) [inline webpack.config.js:208]: We should be able to take this out once we switch to using SFC.

**Review by gary-kim — APPROVED** [review]

File: webpack.config.js (hunk 2 above)
Comment (silverwind) [reply webpack.config.js:208]: What's SFC?

Comment (gary-kim) [reply webpack.config.js:208]: [Single File Components](https://vuejs.org/v2/guide/single-file-components.html), so using `.vue` files. We can use the vue-template-compiler to compile them during the build process so only the runtime is needed.

Comment (silverwind) [reply webpack.config.js]: Ah, I see it's https://vuejs.org/v2/guide/single-file-components.html

Comment (silverwind) [reply webpack.config.js:208]: Indeed, that would be great because it saves a few bytes on Vue itself.

**Comment by silverwind** [comment]
Rebased once more, good to go.

**Pull request merged**